### ISSUE TITLE
Circuit relay sotops accepting connexions

### ISSUE THREAD

**Issue by julienmalard** [libp2p/js-libp2p#2833]
- **Version**: 2.2.1

- **Platform**: Issue encountered on Ubuntu and MacOS

- **Subsystem**: Circuit relay v2

#### Description:
The circuit relay stops accepting new reservations after 5-10 connections by peers. Rebooting the relay temporarily solves the issue. The problem seems to occur independently of whether the reconnecting peers are the same peer or different peers. While peers can always connect to the relay, they are unable to connect to each other through p2p-circuit.

#### Steps to reproduce the error:
1. Start a circuit relay.
2. Connect a few peers to the relay.
3. Disconnect and reconnect 5-10 times.
4. Eventually, peers will be unable to connect to each other through p2p-circuit on the relay.

**Comment by achingbrain** [comment]
There is a [limit](https://github.com/libp2p/js-libp2p/blob/main/packages/transport-circuit-relay-v2/src/server/index.ts#L37-L40) to how many reservations a relay will accept - are you sure you're not hitting this limit?

**Comment by julienmalard** [comment]
Yes, few or no peers are connected to the relay when the issue is encountered. I have a feeling that the reservations or reservation count is not being cleared.

**Comment by achingbrain** [comment]
Can you share a link to a demo project that shows the problem?

**Comment by julienmalard** [comment]
Sure; here's the relay code in question: https://github.com/reseau-constellation/relai-libp2p/
The same problem occurs whether I run it on localhost or properly deployed on DigitalOcean with a domain name and SSL/nginx.
Many thanks!!

**Comment by silkroadnomad** [comment]
I had this problem too. When you develop locally you reach the maximum reservations very quickly. E.g. when using a browser I was ending up to set it to infinity for local developments. When you test in production the same can happen quickly. You can enable libp2p logs in browser and nodejs to watch reservations the issue. https://github.com/silkroadnomad/libp2p-relay/blob/main/relay/src/relay.js#L38 (not sure if thats 100% correct)

**Comment by julienmalard** [comment]
Thanks for confirming @silkroadnomad . However, I still have the issue even with newly generated libp2p nodes which have not yet tried to connect to the relay. Does the node protection track the IP address?

**Comment by silkroadnomad** [comment]
Yes, I guess so. In the beginning, I tried switching off and on the relay, which helped for some time. Maybe deleting the data store etc might be of help in case you have.

Intelligent protection from malicious peers would go beyond the peerId and would track ip addresses and ip networks in worst case. I'd guess that libp2p is so intelligent "by nature." At least I expect the libp2p developers to be. 

I remember having seen a very detailed document about this topic. I can't remember anymore where. I'll post it as soon as I remember. Must be somewhere around here. Maybe documentation of gossip sub might help here.
